pkg/dns/provider/entry.go

```diff
@@ -347,11 +395,17 @@ func (this *EntryVersion) Setup(logger logger.LogContext, state *state, p *Entry
 		this.status.TTL = nil
 	}
 
-	spec = &this.object.DNSEntry().Spec
-
 	///////////// validate
 
-	targets, warnings, verr := validate(state, this)
+	spec, dnsref, targets, warnings, verr := validate(logger, state, this)
```

Comment:
the former spec is used in line 389 to set the TTL. validate must be called before line 379